apiml-common/src/main/java/com/ca/mfaas/product/gateway/GatewayClient.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.product.gateway;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * Container object for {@link GatewayConfigProperties}
+ */
+@Component
+public class GatewayClient {
+
+    private GatewayConfigProperties gatewayConfigProperties;
+
+    public GatewayClient() {
+    }
+
+    public GatewayClient(GatewayConfigProperties gatewayConfigProperties) {
+        this.gatewayConfigProperties = gatewayConfigProperties;
+    }
+
+    public void setGatewayConfigProperties(GatewayConfigProperties gatewayConfigProperties) {
+        this.gatewayConfigProperties = gatewayConfigProperties;
+    }
+
+    /**
+     * Retrieve Gateway configuration
+     * When Gateway configuration is not known, GatewayNotFoundException is thrown
```

Comment:
Maybe "When Gateway is not configured, `GatewayNotFoundException` is thrown"? 
Unless there's a case in which Gateway configuration can be unknown, ignore my comment.

Reply:
it is trying to find the gateway, and before it knows where it is the location of the gateway is not known.
Would you prefer to phrase it as "When Gateway configuration is unknown ... " ?